src/hooks/useSignupHandler.ts

```diff
@@ -0,0 +1,219 @@
+import { useAuth } from "contexts";
+import { db } from "../firebase";
+import { collection, getDocs, query, where } from "firebase/firestore";
+import { FormEvent, useReducer } from "react";
+import { login, signup } from "services";
+import { toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
+
+function useSignupHandler() {
+  const { setAuthToken, setAuthUser } = useAuth();
+  const navigate = useNavigate();
+
+  type AuthFormAndError = {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+    confirmPassword: string;
+  };
+
```

Comment:
Its good that you have switched to typescript for this project

Reply:
Thankyou :)